darts/ad/scorers/scorers.py

```diff
@@ -0,0 +1,848 @@
+            _sanity_check_2series(s1, s2)
+            s1 = self._check_deterministic(s1, "actual_series")
+            s2 = self._check_deterministic(s2, "pred_series")
+            list_fit_series.append(self._diff_series(s1, s2))
+
+        self.fit(list_fit_series)
+        self._fit_called = True
+
+    @abstractmethod
+    def _fit_core(self, series: Any) -> Any:
+        pass
+
+    @abstractmethod
+    def _score_core(self, series: Any) -> Any:
+        pass
+
+    def _diff_sequence(
+        self, list_series_1: Sequence[TimeSeries], list_series_2: Sequence[TimeSeries]
+    ) -> Sequence[TimeSeries]:
+        """Calls the function ``_diff_series()`` on every pair (s1,s2) in the list (list_series_1,list_series_2).
+
+        `list_series_1` and `list_series_2` must have the same length n. Each pair of series in `list_series_1` and
+        `list_series_2` must be of the same length and width/dimension.
+
+        Parameters
+        ----------
+        series_1
+            1st sequence of time series
+        series_2:
+            2nd sequence of time series
+
+        Returns
+        -------
+        TimeSeries
+            Sequence of series of length n
+        """
+
+        return list(map(self._diff_series, list_series_1, list_series_2))
+
+    def _diff_series(self, series_1: TimeSeries, series_2: TimeSeries) -> TimeSeries:
+        """Applies the ``diff_fn`` to the two time series. Converts two time series into 1.
+
+        series_1 and series_2 must:
+            - have a non empty time intersection
+            - be of the same width/dimension W
+
+        Parameters
+        ----------
+        series_1
+            1st time series
+        series_2:
+            2nd time series
+
+        Returns
+        -------
+        TimeSeries
+            series of width/dimension W
+        """
+        series_1, series_2 = _intersect(series_1, series_2)
+
+        if self.diff_fn == "abs_diff":
+            return (series_1 - series_2).map(lambda x: np.abs(x))
+        elif self.diff_fn == "diff":
+            return series_1 - series_2
+        else:
+            # found an non-existent diff_fn
+            raise ValueError(
+                f"Metric should be 'diff' or 'abs_diff', found {self.diff_fn}"
+            )
+
+
+class NLLScorer(NonFittableAnomalyScorer):
+    """Parent class for all LikelihoodScorer"""
+
+    def __init__(self, window) -> None:
+        super().__init__(returns_UTS=False, window=window)
+
+    def _score_core_from_prediction(
+        self,
+        actual_series: TimeSeries,
+        pred_series: TimeSeries,
+    ) -> TimeSeries:
+        """For each timestamp of the inputs:
+            - the parameters of the considered distribution are fitted on the samples of the probabilistic time series
+            - the negative log-likelihood of the determinisitc time series values are computed
+
+        If the series are multivariate, the score will be computed on each width independently.
```

Comment:
What I meant:

If the series is univariate or multivariate, the score will be computed on each component independently.
If it is multivariate, it will also be computed on each width independently.

Reply:
I have changed the comment to "The score will be computed on each component independently. Additionally, if the series is multivariate, the score will be computed on each width independently."